spec/controllers/dashboard_controller_spec.rb

```diff
@@ -421,6 +421,77 @@
     end
   end
 
+  context '#dialog_definition' do
+    before(:each) do
+      allow_any_instance_of(described_class).to receive(:set_user_time_zone)
+      allow(controller).to receive(:check_privileges).and_return(true)
+    end
+
+    let(:klass) { 'someboringclass' }
+    let(:name) { 'a_random_name' }
+
+    context 'existing dialog' do
+      it 'returns json with data' do
+        data = 'bububububuraky'
+        expect(controller).to receive(:load_dialog_definition).and_return(data)
+        post :dialog_definition, :params => {:name => name, :class => klass}
+        expect(response.status).to eq(200)
+        expect(response.body).to include(data)
+      end
+    end
+
+    context 'not existing dialog' do
+      it 'does not find' do
+        expect(controller).to receive(:load_dialog_definition).and_return(nil)
+        post :dialog_definition, :params => {:name => name, :class => klass}
+        expect(response.status).to eq(404)
+      end
+    end
+
+    context 'directory traversal' do
+      it 'ignores anything but [a-z_]' do
+        expect(controller).to receive(:load_dialog_definition).with('abc_de', klass).and_return('data')
+        post :dialog_definition, :params => {:name => '../../abc-!_de', :class => klass}
+      end
+    end
+  end
+
+  context 'private #load_dialog_definition' do
```

Comment:
`describe` and you don't need the private

Reply:
The "private" is part of the description, surely I don't >>need<< it. But I want it.